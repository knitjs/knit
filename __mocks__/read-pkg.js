const rp = jest.genMockFromModule('read-pkg');

let mockPackages = Object.create(null);
function __setMockPackages(newMockPackages) {
  mockPackages = Object.keys(newMockPackages).reduce((acc, p) => ({
    ...acc,
    [p]: newMockPackages[p],
  }), {});
}

function sync(directoryPath) {
  return mockPackages[directoryPath];
}

rp.__setMockPackages = __setMockPackages;
rp.sync = sync;

module.exports = rp;
