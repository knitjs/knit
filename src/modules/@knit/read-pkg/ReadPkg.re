type syncOptions = {normalize: bool};

external sync : Js.String.t => syncOptions => string = "" [@@bs.module "read-pkg"];

let readPkg dir_path package => {
  let ret = sync (PathJoin.pathJoin dir_path package) {normalize: false};
  ()
};