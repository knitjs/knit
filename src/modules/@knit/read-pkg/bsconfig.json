{
  "name": "@knit/read-pkg",
  "sources": ".",
  "package-specs": ["commonjs"],
  "bsc-flags": [
    "-bs-cross-module-opt"
  ],
  "bs-dependencies": [
    "@knit/path-join"
  ],
  "generate-merlin": false
}
