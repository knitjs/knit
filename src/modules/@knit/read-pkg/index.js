/* @flow */

import rp from 'read-pkg';

import pathJoin from '@knit/path-join';

import type { TPkgJson } from '@knit/needle';

type TReadPkg = (d: string, m: string) => TPkgJson;
 const readPkg: TReadPkg = (dir, mod) => {
  const ret = rp.sync(pathJoin(dir, mod), {
    normalize: false,
  });
  if (!ret) {
    throw {
      message: `could not find a \`package.json\` in ${pathJoin(dir, mod)}.`,
      stderr: `All directories under ${dir} are expected to be node modules. To ignore a directory add a 'private: true' to the package.json.`,
    };
  }
  return ret;
};

export default readPkg;