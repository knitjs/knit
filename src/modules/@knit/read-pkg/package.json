{
  "name": "@knit/read-pkg",
  "version": "0.4.1"
}
