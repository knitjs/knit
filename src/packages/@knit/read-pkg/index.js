/* @flow */

import rp from "read-pkg";

import type { TPkgJson } from "@knit/needle";

type TReadPkg = (pkg: {
  path: string,
  workspace: string,
  dir: string
}) => TPkgJson;
const readPkg: TReadPkg = pkg => {
  const ret = rp.sync({
    cwd: pkg.path,
    normalize: false
  });
  if (!ret) {
    throw {
      message: `could not find a \`package.json\` in ${pkg.path}`,
      stderr: `All directories under \`${pkg.workspace}\` are expected to be node modules. To ignore a directory add a 'private: true' to the package.json.`
    };
  }
  return ret;
};

export default readPkg;
