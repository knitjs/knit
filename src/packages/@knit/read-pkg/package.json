{
  "name": "@knit/read-pkg",
  "version": "0.6.0"
}
