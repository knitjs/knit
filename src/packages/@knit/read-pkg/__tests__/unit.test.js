/* flow */

const pathJoin = require("@knit/path-join");

const rp = require("..");

describe("readPkg", () => {
  it("returns json if package exists", () => {
    require("read-pkg").__setMockPackages({
      [pathJoin("ws", "@scope", "package")]: { name: "@scope/package" }
    });
    expect(
      rp({
        path: pathJoin("ws/@scope/package"),
        workspace: "ws",
        dir: "@scope/package"
      })
    ).toEqual({ name: "@scope/package" });
  });
  it("throws if package is missing", () => {
    expect(() => rp({ path: "DNE" })).toThrowErrorMatchingSnapshot();
  });
});
