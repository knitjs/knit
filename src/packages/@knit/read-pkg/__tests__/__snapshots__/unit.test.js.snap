// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readPkg throws if package is missing 1`] = `"could not find a \`package.json\` in DNE."`;
